lib/src/map/built_map.dart

```diff
@@ -30,20 +30,46 @@ abstract class BuiltMap<K, V> {
   ///
   /// Wrong: `new BuiltMap({1: '1', 2: '2', 3: '3'})`.
   ///
-  /// Right: `new BuiltMap<int, String>({1: '1', 2: '2', 3: '3'})`,
+  /// Right: `new BuiltMap<int, String>({1: '1', 2: '2', 3: '3'})`.
   ///
   /// Rejects nulls. Rejects keys and values of the wrong type.
   factory BuiltMap([map = const {}]) {
     if (map is _BuiltMap && map.hasExactKeyAndValueTypes(K, V)) {
       return map as BuiltMap<K, V>;
     } else if (map is Map || map is BuiltMap) {
-      return new _BuiltMap<K, V>.copyAndCheck(map.keys, (k) => map[k]);
+      return new _BuiltMap<K, V>.copyAndCheckTypes(map.keys, (k) => map[k]);
     } else {
       throw new ArgumentError(
           'expected Map or BuiltMap, got ${map.runtimeType}');
     }
   }
 
+  /// Instantiates with elements from a [Map].
+  ///
+  /// Must be called with a generic type parameter.
+  ///
+  /// Wrong: `new BuiltMap.from({1: '1', 2: '2', 3: '3'})`.
+  ///
+  /// Right: `new BuiltMap<int, String>.from({1: '1', 2: '2', 3: '3'})`.
+  ///
+  /// Rejects nulls. Rejects keys and values of the wrong type.
```

Comment:
Why does it say `copyAndCheckTypes rejects null` and the implementation doesn't check for null?

Reply:
Checking for types also checks for null--since `null is Foo` always returns `false`.